x-pack/plugins/ingest_manager/server/services/epm/kibana/assets/install.ts

```diff
@@ -75,24 +107,47 @@ export const deleteKibanaInstalledRefs = async (
   });
 };
 export async function getKibanaAssets(paths: string[]) {
-  const isKibanaAssetType = (path: string) => Registry.pathParts(path).type in KibanaAssetType;
-  const filteredPaths = paths.filter(isKibanaAssetType);
-  const kibanaAssets = await Promise.all(filteredPaths.map((path) => getKibanaAsset(path)));
-  return kibanaAssets;
+  const kibanaAssetTypes = Object.values(KibanaAssetType);
+  const isKibanaAssetType = (path: string) => {
+    const parts = Registry.pathParts(path);
+
+    return parts.service === 'kibana' && (kibanaAssetTypes as string[]).includes(parts.type);
+  };
+
+  const filteredPaths = paths
+    .filter(isKibanaAssetType)
+    .map<[string, AssetParts]>((path) => [path, Registry.pathParts(path)]);
+
+  const assetArrays: Array<Promise<ArchiveAsset[]>> = [];
+  for (const assetType of kibanaAssetTypes) {
+    const matching = filteredPaths.filter(([path, parts]) => parts.type === assetType);
+
+    assetArrays.push(Promise.all(matching.map(([path]) => path).map(getKibanaAsset)));
+  }
+
+  const resolvedAssets = await Promise.all(assetArrays);
+
+  const result = {} as Record<KibanaAssetType, ArchiveAsset[]>;
```

Comment:
Mind moving `Record<KibanaAssetType, ArchiveAsset[]>` to the function return type?

Reply:
Can we remove the remove the `as Record...`, leaving either `const result = {}` or 
```suggestion
  const result: Record<KibanaAssetType, ArchiveAsset[]> = {};
```